x/concentrated-liquidity/types/keys.go

```diff
@@ -198,3 +207,18 @@ func KeyUptimeAccumulator(poolId uint64, uptimeIndex uint64) string {
 	uptimeIndexStr := strconv.FormatUint(uptimeIndex, uintBase)
 	return strings.Join([]string{string(UptimeAccumulatorPrefix), poolIdStr, uptimeIndexStr}, "/")
 }
+
+// Helper Functions
+
+func MustGetPoolIdFromShareDenom(denom string) uint64 {
+	parts := strings.Split(denom, "/")
+	if len(parts) != 4 {
+		panic("invalid share denom")
+	}
+	poolIdStr := parts[2]
+	poolId, err := strconv.Atoi(poolIdStr)
+	if err != nil {
+		panic(err)
+	}
+	return uint64(poolId)
+}
```

Comment:
It might also be useful to validate that denom matches the desired prefix

Reply:
added prefix validation here https://github.com/osmosis-labs/osmosis/pull/4882/commits/cc1624e9e5898a5ad8026bdd9f31a71e4851aa19